File: CS-Examples/13_MailMerge/NestedMailMerge.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Windows.Forms;
using Spire.Doc;

namespace NestedMailMerage
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            // Create a list to store DictionaryEntry objects
			List<DictionaryEntry> list = new List<DictionaryEntry>();

			// Create a DataSet object
			DataSet dsData = new DataSet();

			// Read XML data into the DataSet
			dsData.ReadXml(@"..\..\..\..\..\..\Data\Orders.xml");

			// Create a Document object
			Document document = new Document();

			// Load a Word document from file
			document.LoadFromFile(@"..\..\..\..\..\..\Data\NestedMailMerge.doc");

			// Create a DictionaryEntry for "Customer" with an empty value and add it to the list
			DictionaryEntry dictionaryEntry = new DictionaryEntry("Customer", string.Empty);
			list.Add(dictionaryEntry);

			// Create a DictionaryEntry for "Order" with a nested region condition and add it to the list
			dictionaryEntry = new DictionaryEntry("Order", "Customer_Id = %Customer.Customer_Id%");
			list.Add(dictionaryEntry);

			// Execute mail merge with nested regions using the DataSet and list of DictionaryEntry objects
			document.MailMerge.ExecuteWidthNestedRegion(dsData, list);

			// Save the merged document to a file 
			document.SaveToFile("Sample.docx", FileFormat.Docx);

			// Dispose the Document object 
			document.Dispose();

            //Launching the MS Word file.
            WordDocViewer("Sample.docx");
        }

        private void WordDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
    }
}

File: VB-Examples/13_MailMerge/NestedMailMerge.vb
Imports System.Collections
Imports Spire.Doc

Namespace NestedMailMerage
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			'Create a list to store DictionaryEntry objects
			Dim list As New List(Of DictionaryEntry)()

			'Create a DataSet object
			Dim dsData As New DataSet()

			'Read XML data into the DataSet
			dsData.ReadXml("..\..\..\..\..\..\Data\Orders.xml")

			'Create a Document object
			Dim document As New Document()

			'Load a Word document from file
			document.LoadFromFile("..\..\..\..\..\..\Data\NestedMailMerge.doc")

			'Create a DictionaryEntry for "Customer" with an empty value and add it to the list
			Dim dictionaryEntry As New DictionaryEntry("Customer", String.Empty)
			list.Add(dictionaryEntry)

			'Create a DictionaryEntry for "Order" with a nested region condition and add it to the list
			dictionaryEntry = New DictionaryEntry("Order", "Customer_Id = %Customer.Customer_Id%")
			list.Add(dictionaryEntry)

			'Execute mail merge with nested regions using the DataSet and list of DictionaryEntry objects
			document.MailMerge.ExecuteWidthNestedRegion(dsData, list)

			'Save the merged document to a file 
			document.SaveToFile("Sample.docx", FileFormat.Docx)

			'Dispose the document
			document.Dispose()

			'Launching the MS Word file.
			WordDocViewer("Sample.docx")
		End Sub

		Private Sub WordDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
	End Class
End Namespace
